### DISCUSSION TITLE
Crediantials in Yarp

### DISCUSSION THREAD

**Discussion by mayuriswan** [dotnet/yarp#2290] [Q&A] [answered]
### Why is this important to you?

hey i m trying to pass some credentials in the request header to auto authenticate to another power Pi server  but it keeps asking me for credentials 
`using System.Net;
using System.Net.Http.Headers;
using System.Text;

var builder = WebApplication.CreateBuilder(args);
builder.Services.AddReverseProxy()
    .LoadFromConfig(builder.Configuration.GetSection("ReverseProxy"))
        .ConfigureHttpClient((_, handler) => handler.ActivityHeadersPropagator = null);

var app = builder.Build();
app.UseMiddleware<ProxyMiddleware>();

app.MapReverseProxy();

app.UseMiddleware<ProxyMiddleware>();

app.Run(); `

**Comment by Tratcher** [comment, marked as answer]
How are you setting the credentials? (Don't show us the actual credentials 😆).

I suggest adding the header directly to the outgoing HttpRequestMessage using a request transform.
https://microsoft.github.io/reverse-proxy/articles/transforms.html#addrequesttransform
https://github.com/microsoft/reverse-proxy/blob/27554a39d5417b32098d8b216f287a48b585789d/samples/ReverseProxy.Transforms.Sample/Startup.cs#L48-L55
